data/profiles/taskrunner.py

```diff
@@ -1,61 +1,153 @@
-#!/usr/bin/python
-
 # Copyright 2016, EMC, Inc.
 
 """
 Download and execute python scripts from the RackHD tasks api in a loop
 """
 
-import requests
 import json
 import sys
 from time import sleep
 
+class Cisco(object):
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def download_task_data():
+        """
+        Download a python script from the tasks API
+        """
+        _task_data = urllib2.urlopen(TASKS_URI).read()
+        poc_log("Downloading task data")
+        poc_log("_task_data: {0}".format(_task_data))
+
+        return json.loads(_task_data)
+
+    @staticmethod
+    def download_script(_downloadUrl):
+        """
+        Download a python script from the tasks API
+        """
+        url = urllib2.urlparse.urljoin('http://<%=server%>:<%=port%>', _downloadUrl)
+        poc_log("Downloading script at {}".format(url))
+        script = urllib2.urlopen(url).read()
+        poc_log("{0}".format(script))
+        with open('script.py', 'w') as rackhd_script:
+            rackhd_script.write(script)
+
+    @staticmethod
+    def post_task_data(_task_data, _json_content_type):
+        poc_log("Posting task data\n {}".format(_task_data))
+        _task_data = json.dumps(_task_data)
+        _req = urllib2.Request(TASKS_URI, _task_data, _json_content_type)
+        return _req
+
+    @staticmethod
+    def post_task_data_response(_req):
+        poc_log("Posting task data response\n")
+        urllib2.urlopen(_req)
+
+
+class Brocade(object):
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def download_task_data():
+        """
+        Download a python script from the tasks API
+        """
+        poc_log("Downloading task data")
+        session = requests.Session()
+        response = session.get(TASKS_URI, auth=("", ""))
+        _task_data = response.text
+        session.close()
+        poc_log("_task_data: {0}".format(_task_data))
+        return json.loads(_task_data)
+
+    @staticmethod
+    def download_script(_downloadUrl):
+        """
+        Download a python script from the tasks API
```

Comment:
This isn't really using the tasks API is it?  Should this say "from downloadUrl"?